src/wally-connector.ts

```diff
@@ -234,96 +243,168 @@ class WallyConnector {
+   */
+  async request<T extends MethodNameType>(
+    req: RequestObj<T>
+  ): Promise<MethodResponse<T> | null> {
     if (!this.isLoggedIn()) {
       await this.loginWithEmail();
     }
 
-    switch (req.method) {
-      case MethodName.REQUEST_ACCOUNTS:
-        return this.requestAccounts();
-      // TODO: figure out which name to use
-      case MethodName.PERSONAL_SIGN:
-      case MethodName.SIGN:
-        return this.signMessage(req.params);
-      case MethodName.GET_BALANCE:
-        return this._request(req.method, req.params);
+    if (Object.values(WallyMethodName).indexOf(req.method as any) > -1) {
+      return this.requestWally(
+        req.method as WallyMethodName,
+        'params' in req ? (req.params as WallyMethodParams<T>) : undefined
+      ) as Promise<WallyResponse<T>>;
+    } else if (Object.values(RPCMethodName).indexOf(req.method as any) > -1) {
+      return this.requestRPC(
+        req.method as RPCMethodName,
+        'params' in req ? (req.params as RPCMethodParams<T>) : undefined
+      );
+    } else {
+      throw new Error(`Method ${req.method} is unsupported at this time.`);
+    }
+  }
+
+  private formatWallyParams<T extends WallyMethodName>(
+    method: T,
+    params: WallyMethodParams<T>
+  ): string {
+    if (method === WallyMethodName.SIGN) {
+      return JSON.stringify({ message: (params as SignParams)[1] });
+    } else if (method === WallyMethodName.PERSONAL_SIGN) {
+      return JSON.stringify({ message: (params as PersonalSignParams)[0] });
+    } else {
+      return JSON.stringify(params);
+    }
+  }
+
+  private formatWallyResponse<T extends WallyMethodName>(
+    method: T,
+    data: any
+  ): WallyResponse<T> | null {
+    switch (method) {
+      case WallyMethodName.ACCOUNTS:
+      case WallyMethodName.REQUEST_ACCOUNTS: {
+        const { address } = data;
+        this.selectedAddress = address;
+        return [address] as WallyResponse<T>;
+      }
+      case WallyMethodName.SIGN:
+      case WallyMethodName.PERSONAL_SIGN:
+      case WallyMethodName.SIGN_TRANSACTION:
+      case WallyMethodName.SIGN_TYPED: {
+        const { signature } = data;
+        return signature;
+      }
     }
+    return null;
   }
 
-  async requestAccounts(): Promise<string[]> {
+  /**
+   * Method used doing wallet-related actions like requesting accounts
+   * and signing things - actions that require wallet/private key access
+   * and are basically the core wally value prop.
+   * @param method The RPC method name associated with the wally api call
+   * @param params The json rpc spec params (*not* wally's spec)
+   * @returns WallyResponse - adheres to the json rpc spec
+   */
+  private async requestWally<T extends WallyMethodName>(
+    method: T,
+    params: WallyMethodParams<T>
+  ): Promise<WallyResponse<T> | null> {
     let resp: Response;
     try {
-      resp = await fetch(`${this.host}/oauth/me`, {
+      resp = await fetch(`${this.host}/oauth${WALLY_ROUTES[method]}`, {
```

Comment:
is this missing a `/` between oauth and the method route?

Reply:
no, but it doesn't look right, huh? 😅  i'll move the leading slashes out of the constants to here instead ⚡